Add infoblox DNS mgmt

lomeroe opened the pull request:
Module and state for adding/removing/updating DNS records on an infoblox DNS system

Resolves #27547

jfindlay commented on salt/modules/infoblox.py:
```diff
@@ -0,0 +1,479 @@
+# -*- coding: utf-8 -*-
+'''
+Module for managing Infoblox
+
```
`.. versionadded:: Boron`

lomeroe committed:
lint fixes

lomeroe committed:
trying to make lint happy...

lomeroe committed:
lint fix, not sure why I was passing in an empty string as the 2nd value :/

rallytime commented:
Thanks for adding this new module and state! Looks great! 

There are a couple of more items that need to be cleaned up, as you can see in the [test failures](https://jenkins.saltstack.com/job/salt-pr-rs-cent7-n/8465/testReport/junit/integration.modules.sysmod/SysModuleTest/test_valid_docs/).

Each new function should have a docstring (looks like you're good there) as well as a CLI example (or a state example would be most excellent as well for the state functions). Some example syntax for that typically looks like this:

Execution Modules:
```
CLI Example:

.. code-block:: bash

    salt my-minion module.function arg1 arg2 kwarg1=somekwarg
```
States:
```
Example:

.. code-block:: yaml

    state-id:
      state.function:
        - name: my-state
        - kwarg1: value
        - kwarg1: value
```
In addition to adding those examples, it'd be also most excellent if you could add some version added directives at the top of each new file (where @jfindlay suggested in your new module) and also add that to the state.

Once that is cleaned up, we'd be happy to get these merged. Thanks again!

lomeroe committed:
Documentation updates with examples

version added string
one module modification to standardize record_type parameter name

rallytime commented:
Great! Thanks!
